elasticdl/python/tests/layer_test.py

```diff
@@ -160,11 +165,13 @@ def test_embedding_layer_with_input_length(self):
         )
         ids = [[0, 1, 3, 8], [5, 3, 2, 3]]
         flatten_ids = ids[0] + ids[1]
-        values = layer.call(ids)
-        values = values.numpy().reshape(-1, output_dim)
-        for index, idx in enumerate(flatten_ids):
-            correct_value = np.array([idx] * output_dim, dtype=np.float32)
-            self.assertTrue((values[index] == correct_value).all())
+        call_fns = [lambda: layer.call(ids), lambda: module_call(layer, ids)]
+        for call_fn in call_fns:
+            values = call_fn()
+            values = values.numpy().reshape(-1, output_dim)
+            for index, idx in enumerate(flatten_ids):
+                correct_value = np.array([idx] * output_dim, dtype=np.float32)
+                self.assertTrue((values[index] == correct_value).all())
```

Comment:
Same here and also other places (double check though)

Reply:
Remove most of repeat code. Leave two test cases unchanged. Because removing a small quantity of repeat code in these two function requires changing the logic of test case itself. (`test_embedding_layer_gradient` and `test_embedding_layer_gradient_with_sparse_inputs`)